3-UsingAPIs/Week2/homework/ex1-programmerFun/index.js

```diff
@@ -15,29 +16,37 @@
    url with `.shx`. There is no server at the modified url, therefore this 
    should result in a network (DNS) error.
 ------------------------------------------------------------------------------*/
-function requestData(url) {
-  // TODO return a promise using `fetch()`
+async function requestData(url) {
+  const data = await fetch(url);
+  const response = await data.json();
+  if (response.status <= 200 && response.status >= 400) {
```

Comment:
This should probably be `< 200`, right? Now the function will fail if it receives a 200 OK status code. The reason that the function works with the xkcd url below, is because that website apparently returns a 304, but 200 would have been reasonable too.